share/wake/lib/core/list.wake

```diff
@@ -71,7 +71,7 @@ export def tail: List a => List a = match _
 #
 #   map str    (seq 5) = "0", "1", "2", "3", "4", Nil
 #   map (_+10) (seq 5) = 10, 11, 12, 13, 14, Nil
-export def map (f: a => b): (List a => List b) =
+export def map (f: a => b): List a => List b =
```

Comment:
Any reason for removing the `()`s?
They were there to indicate the idea that map turns a function on elements into a function on lists.

Reply:
No specific reason beyond the fact that this file is the only one which includes the parentheses; I assumed it was an early implementation of the type annotation syntax which was never updated.  I'm perfectly happy to amend the commit to leave them in, but I will say that (A), the extension hover info boxes don't include the parentheses even if the definition does, and (B), with my familiarity with functional programming, I assume the connotations of that currying even without it being called out.

Reply:
I guess it makes less sense now than when it was typed as `(a => b) => (List a => List b)`.

The hover extension does not show the `()`s, because it reformats (with minimal `()`s) the unified final type.